.github/workflows/push_dev_images.yml

```diff
@@ -0,0 +1,37 @@
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - 'main'
+      - 'releases/*'
+    types:
+      - closed
+
+name: Push dev environment images and update caches
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.base.ref }}
+  cancel-in-progress: true
+
+env:
+  registry: ghcr.io
+  image_name: nvidia/cuda-quantum-devdeps
+
+jobs:
+  devdeps_gcc11:
+    uses: ./.github/workflows/dev_environment.yml
+    if: github.event.pull_request.merged == true
+    with:
+      toolchain: gcc11
+      cache_name: cuda-quantum-devdeps-gcc11
```

Comment:
Won't you cache collision because its using the same cache name for the main branch and all release branches?

Reply:
Ups, yes. Updated, and I cleaned up the arguments to the workflow, which now takes the prefix for the cache as argument (github.ref_name by default and set to github.event.pull_request.base.ref by push_dev_images), but handles everything else, including looking at the right cache to pull from when the pr specific cache does not exist yet.